crates/rolldown/src/ast_scanner/side_effect_detector.rs

```diff
@@ -194,10 +206,64 @@ impl<'a> SideEffectDetector<'a> {
       | Expression::YieldExpression(_)
       | Expression::JSXElement(_)
       | Expression::JSXFragment(_) => true,
+      Expression::CallExpression(expr) => {
+        let leading_comment_index: Option<usize> = if self.attached_comment_vecmap.is_empty() {
+          None
+        } else {
+          // because the span is the content of comment, so the two span should never overlapped,
+          // the result Variant should be `Err(usize)`
+          let insert_index = self
+            .attached_comment_vecmap
+            .binary_search_by(|probe| probe.0.end.cmp(&expr.span.start));
```

Comment:
I'm fine with this implementation.

However, this feel not right about how to get comments of ast nodes. I was expecting we could get comment of ast node by simply using API like `trivias.get_comments(expr.span)`.

Once oxc support emit comments in codegen, we could see how oxc get comments from oxc ast nodes.

@Boshen cc